recipes/lora_finetune_single_device.py

```diff
@@ -245,8 +244,12 @@ def _setup_model(
         if lora_weights_state_dict:
             model.load_state_dict(lora_weights_state_dict, strict=False)
 
-        # Validate model was loaded in with the expected dtype.
-        utils.validate_expected_param_dtype(model, dtype=self._dtype)
+        # Validate model adapter params were loaded in with the expected dtype
+        # TODO (rohan-varma): Further validation to ensure the appropriate base params
+        # are NF4 vs bf16 based on the quantization config.
+        utils.validate_expected_param_dtype(
+            self.adapter_params.items(), dtype=self._dtype
+        )
```

Comment:
Why was `validate_expected_param_dtype` passing before? Shouldn't it have contained NF4 dtypes?

Reply:
Before in torchao, nf4.dtype would give torch.bf16 so this passed. now nf4.dtype gives `float8e5m2`